### PR TITLE
View controller docs

### PR DESCRIPTION
**Pull request by squirrelo**
Updates view-controller.js to be gjslint and jsdoc compatible.

### CODE REVIEW

File: emperor/support_files/js/view-controller.js [hunk 1]
```diff
@@ -5,62 +5,75 @@ define([
...
+   * @constructs EmperorViewControllerABC
    *
    */
   function EmperorViewControllerABC(container, title, description) {
+    /**
+     * @type {Node}
+     * jQuery element for the parent container.
+     */
     this.$container = $(container);
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: Can you add another `*`, it seems like that's what marks the block as a jsdoc annotation (at least in the syntax highlighter).

File: emperor/support_files/js/view-controller.js [hunk 2]
```diff
@@ -5,62 +5,75 @@ define([
...
+     * @type {String}
+     * Human-readable title of the tab.
+     */
     this.title = title;
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: ☝️

File: emperor/support_files/js/view-controller.js [hunk 3]
```diff
@@ -5,62 +5,75 @@ define([
...
+     * @type {String}
+     * Human-readable description of the tab.
+     */
     this.description = description;
 
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: Can you add the missing `*`.

File: emperor/support_files/js/view-controller.js [hunk 4]
```diff
@@ -5,62 +5,75 @@ define([
...
+     * displayed in tab. This goes below the header.
+     */
     this.$body = null;
+    /**
+     * @type {Node}
+     * jQuery element for the header which contains the uppermost elements
+     * displayed in a tab.
+     */
     this.$header = null;
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: Can you add the missing `*`.

File: emperor/support_files/js/view-controller.js [hunk 5]
```diff
@@ -243,15 +241,33 @@ define([
...
 
+    /**
+     * @type {Object}
+     * This is object is keyed by unique identifiers and the values are
+     * DecompositionView objects referring to a set of objects presented on
+     * screen. This dictionary will usually be shared by all the tabs in the
+     * application. This argument is passed by reference.
+     */
     this.decompViewDict = decompViewDict;
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: Can you add the missing `*`.

File: emperor/support_files/js/view-controller.js [hunk 6]
```diff
@@ -360,6 +378,11 @@ define([
       columns.unshift(options.slickGridColumn);
     }
 
+    /*
```
Comment (ElDeveloper) [inline emperor/support_files/js/view-controller.js]: The alignment of this whole block looks a bit off.

**Comment by ElDeveloper** [comment]
👍 just a few minor comments.

File: emperor/support_files/js/view-controller.js (hunk 1 above)
Comment (squirrelo) [inline emperor/support_files/js/view-controller.js]: fixed here and all subsequent ones.

File: emperor/support_files/js/view-controller.js (hunk 6 above)
Comment (squirrelo) [inline emperor/support_files/js/view-controller.js]: fixed

**Comment by squirrelo** [comment]
Comments addressed

**Comment by ElDeveloper** [comment]
Seems like Travis is having issues with builds, lets' wait until those get cleared:

**Comment by ElDeveloper** [comment]
https://www.traviscistatus.com

**Pull request merged**